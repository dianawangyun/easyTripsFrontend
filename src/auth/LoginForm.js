import React, { useState } from "react";
import { useHistory } from "react-router-dom";

function LoginForm({ login }) {
    const history = useHistory();
    const [formData, setFormData] = useState({
        username: "",
        password: "",
    });
    const [formErrors, setFormErrors] = useState([]);

    async function handleSubmit(e) {
        e.preventDefault();
        let res = await login(formData);
        if (res.success) {
            history.push("/home");
        } else {
            setFormErrors(res.errs);
        }
    }

    function handleChange(evt) {
        const { name, value } = evt.target;
        setFormData((data) => ({ ...data, [name]: value }));
    }

    return (
        <div className="container">
            <h2 className="mt-5 mb-4 text-center">Log In</h2>
            <div className="row d-flex justify-content-center">
                <div className="col-md-6">
                    <form onSubmit={handleSubmit}>
                        <div className="form-group">
                            <label htmlFor="username">Username: </label>
                            <input
                                name="username"
                                id="login-username"
                                className="form-control"
                                value={formData.username}
                                onChange={handleChange}
                                required
                                type="text"
                            />
                        </div>
                        <div className="form-group">
                            <label htmlFor="login-password">Password: </label>
                            <input
                                type="password"
                                name="password"
                                className="form-control"
                                value={formData.password}
                                onChange={handleChange}
                                required
                            />
                        </div>
                        {formErrors.length ? (
                            <ul>
                                {formErrors.map((e) => (
                                    <li className="text-danger">{e}</li>
                                ))}
                            </ul>
                        ) : null}
                        <div className="text-center">
                            <button
                                className="btn btn-success mb-5"
                                type="submit"
                            >
                                Log In
                            </button>
                        </div>
                    </form>
                </div>
            </div>
        </div>
    );
}

export default LoginForm;
