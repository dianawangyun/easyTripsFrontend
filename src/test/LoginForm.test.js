import React from "react";
import { render, fireEvent } from "@testing-library/react";
import LoginForm from "../auth/LoginForm";
import { MemoryRouter } from "react-router";

it("renders without crashing", function () {
    render(<LoginForm />);
});

it("matches snapshot", function () {
    const { asFragment } = render(
        <MemoryRouter>
            <LoginForm />
        </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});

it("should show login form", function () {
    const { queryByText } = render(<LoginForm />);
    expect(queryByText("Username:")).toBeInTheDocument();
});

it("can type in the input", function () {
    const login = () => {
        return { success: true };
    };
    const { getByTestId } = render(<LoginForm login={login} />);
    const userName = getByTestId("login-username");
    fireEvent.change(userName, { target: { value: "test" } });
    expect(userName.value).toEqual("test");
});
