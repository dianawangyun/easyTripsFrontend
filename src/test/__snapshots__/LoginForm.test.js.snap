// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <h2
      class="mt-5 mb-4 text-center"
    >
      Log In
    </h2>
    <div
      class="row d-flex justify-content-center"
    >
      <div
        class="col-md-6"
      >
        <form>
          <div
            class="form-group"
          >
            <label
              for="username"
            >
              Username: 
            </label>
            <input
              class="form-control"
              data-testid="login-username"
              id="login-username"
              name="username"
              required=""
              type="text"
              value=""
            />
          </div>
          <div
            class="form-group"
          >
            <label
              for="login-password"
            >
              Password: 
            </label>
            <input
              class="form-control"
              data-testid="login-password"
              name="password"
              required=""
              type="password"
              value=""
            />
          </div>
          <div
            class="text-center"
          >
            <button
              class="btn btn-success mb-5"
              data-testid="submit"
              type="submit"
            >
              Log In
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
